src/clients/c++/library/grpc_client.cc

```diff
@@ -851,6 +862,13 @@ InferenceServerGrpcClient::AsyncInfer(
   for (const auto& it : headers) {
     async_request->grpc_context_.AddMetadata(it.first, it.second);
   }
+
+  if (options.client_timeout_ != 0) {
```

Comment:
What does the client see when a timeout happens? Do they get an exception? A particular error code returned in the result object?

Reply:
They will see an InferenceServerException with message "Deadline Exceeded" message.

Reply:
Add this to the documentation for "client_timeout".